feat: add vega interpreter as dependency and support custom interpreter

domoritz opened the pull request:
New version of #475
<!-- GITHUB_RELEASE PR BODY: canary-version -->
<details>
  <summary>📦 Published PR as canary version: <code>6.18.3--canary.747.450ca4f.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install vega-embed@6.18.3--canary.747.450ca4f.0
  # or 
  yarn add vega-embed@6.18.3--canary.747.450ca4f.0
  ```
</details>
<!-- GITHUB_RELEASE PR BODY: canary-version -->

domoritz commented:
@stefanw could you review this pull request?

stefanw commented:
> Setting the interpreter sets `ast` to `true` by default.

Currently, this doesn't seem to be included.

I'm unsure what level of abstraction `vega-embed` is going for. An idea could be offer an option `noeval` (or similar) which could then set `ast` and `expr` appropriately behind the scenes. This makes it easy to configure CSP-compliance without having to learn about how `vega` actually achieves it. Or rephrased as a `vega-embed` user, do I care that `vega` needs to generate an AST and pass it to an expression interpreter?

domoritz commented:
> Setting the interpreter sets ast to true by default.

Right, I removed that. I updated the comment.

Since `expr` is already set by default, setting `ast` is pretty much your proposed `noeval`. I agree that `ast` isn't at the right level of thinking for most users but since we already have the option, I can't remove it without making a breaking change.

We can reconsider the naming in the future but for now, I'll leave the names as they are. Hopefully, the documentation is good enough so that people find the option they need.
